tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/03_productsStock/07_enableDeliveryTimeOutOfStockProducts.js

```diff
@@ -0,0 +1,161 @@
+      await this.pageObjects.productsPage.goToAddProductPage();
+      const validationMessage = await this.pageObjects.addProductPage.createEditBasicProduct(productData);
+      await expect(validationMessage).to.equal(this.pageObjects.addProductPage.settingUpdatedMessage);
+    });
+  });
+
+  describe('Enable delivery time out-of-stock', () => {
+    it('should go to \'Shop parameters > Product Settings\' page', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goToProductSettingsPage', baseContext);
+      await this.pageObjects.boBasePage.goToSubMenu(
+        this.pageObjects.boBasePage.shopParametersParentLink,
+        this.pageObjects.boBasePage.productSettingsLink,
+      );
+      const pageTitle = await this.pageObjects.productSettingsPage.getPageTitle();
+      await expect(pageTitle).to.contains(this.pageObjects.productSettingsPage.pageTitle);
+    });
+
+    const tests = [
+      {args: {action: 'enable', enable: true, deliveryTimeText: '8-9 days'}},
+      {args: {action: 'disable', enable: false, deliveryTimeText: ''}},
+    ];
+    tests.forEach((test, index) => {
+      describe(`Check delivery time of out-of-stock products ${test.args.enable} status`, async () => {
+        it(`should ${test.args.action} delivery time of out-of-stock products in BO`, async function () {
+          await testContext.addContextItem(this, 'testIdentifier', `${test.args.action}StockManagement`, baseContext);
+          await this.pageObjects.productSettingsPage.setAllowOrderingOutOfStockStatus(test.args.enable);
+          const result = await this.pageObjects.productSettingsPage.setDeliveryTimeOutOfStock(
+            test.args.deliveryTimeText,
+          );
+          await expect(result).to.contains(this.pageObjects.productSettingsPage.successfulUpdateMessage);
+        });
+
+        it('should view my shop', async function () {
+          await testContext.addContextItem(this, 'testIdentifier', `viewMyShop${index}`, baseContext);
+          page = await this.pageObjects.productSettingsPage.viewMyShop();
+          this.pageObjects = await init();
+          await this.pageObjects.foHomePage.changeLanguage('en');
+          const isFoHomePage = await this.pageObjects.foHomePage.isHomePage();
+          await expect(isFoHomePage, 'Fail to open FO home page').to.be.true;
+        });
+
+        it('should check delivery time block visibility', async function () {
+          await testContext.addContextItem(this, 'testIdentifier', `deliveryTimeBlockVisible${index}`, baseContext);
+          await this.pageObjects.foHomePage.searchProduct(productData.name);
+          await this.pageObjects.searchResultsPage.goToProductPage(1);
+          const isDeliveryTimeBlockVisible = await this.pageObjects.foProductPage.isDeliveryInformationVisible();
+          await expect(isDeliveryTimeBlockVisible).to.equal(test.args.enable);
+        });
+
+        if (test.args.enable) {
```

Comment:
If it is disabled, it should check that no text is present

Reply:
In the step before, we check if element is visible.
if enabled => element visible and we can check text of the element
if disabled => element not visible and we can't check text of undefined.

Reply:
:ok_hand: Misread ;)